packages/web3-eth/src/utils/detect_transaction_type.ts

```diff
@@ -110,43 +114,37 @@ export const defaultTransactionTypeParser: TransactionTypeParser = transaction =
 		return '0x1';
 	}
 
-	// We don't return 0x0 here, because if gasPrice is not
-	// undefined, we still don't know if the network
-	// supports EIP-2718 (https://eips.ethereum.org/EIPS/eip-2718)
-	// and whether we should return undefined for legacy txs,
-	// or type 0x0 for legacy txs post EIP-2718
-	if (!isNullish(tx.gasPrice)) {
-		validateTxTypeAndHandleErrors(transactionType0x0Schema, tx, '0x0');
-	}
```

Comment:
if tx.gasPrice is explicitly defined in user input it should take precedence before using HFs for decision.